src/citrine/_session.py

```diff
@@ -79,7 +79,12 @@ def checked_request(self, method: str, path: str,
             logger.debug('\t{}: {}'.format(k, v))
         logger.debug('END request details.')
 
-        response = super().request(method, uri, **kwargs)
+        try:
+            response = super().request(method, uri, **kwargs)
+        except (ConnectionError, ConnectionResetError):
+            logger.debug('Connection Error, creating a new session')
+            super().__init__()
```

Comment:
What exactly does this do? Does this reset the state of the citrine session? How are the parameters that were used to instantiate the citrine session (e.g. scheme, host, api key) persisted?

Reply:
The citrine-python Session is a subclass of Requests.Session.  The CP Session.__init__() calls the superclass's (Requests.Session's) __init__() with no arguments.  I had a look as Request's Session.__init__() and this is where urllib3 connections are created.

All of the arguments passed to CP's Session.__init__() are left unchanged so they should persist as normal.  I'm not sure which attributes are used by the superclass, e.g. refresh_token.

This PR doesn't try to refresh the token if one of these errors are seen.  This is something we can look into if required.

This change has not been tested with Jupyter notebooks to see if this addresses the possible timeout issue.

Reply:
Got it. This sounds great to me. I'll merge.